lib/BackendClient.js

```diff
@@ -3,9 +3,103 @@
  */
 
 var Q = require('q');
+var _ = require('underscore');
+var kalturaVO = require('./kaltura-client-lib/KalturaVO');
+var kalturaClient = require('./kaltura-client-lib/KalturaClient');
+var kalturaClientBase = require('./kaltura-client-lib/KalturaClientBase');
+var kalturaTypes = require('./kaltura-client-lib/KalturaTypes');
+var config = require('./Configuration');
+var logger = require('./logger/logger')(module);
 
-module.exports = {
-    getLiveEntriesForMediaServer : function(){
-        Q.reject(new Error("Not yet implemented"));
-    }
-};
\ No newline at end of file
+var BackendClient = (function(){
+
+    var createClientConfig = function () {
+        var conf = new kalturaClientBase.KalturaConfiguration(partnerId);
+        conf.serviceUrl = config.get('serviceUrl');
+        return conf;
+    };
+
+    var convertToMs = function (minutes) {
+        return minutes * 60 * 1000;
+    };
+
+    var lastSessionTime = null;
+    var adminSecret = config.get('adminSecret');
+    var partnerId = config.get('partnerId');
+    var mediaServerHostname = config.get('mediaServerHostname');
+    var client = null;
+    var clientConfig = createClientConfig();
+    var sessionDuration = convertToMs(config.get('backendClientSessionDurationMinutes'));
+
+    var BackendClient = {};
+
+    var createSession = function () {
+
+        var deferred = Q.defer();
+        client = new kalturaClient.KalturaClient(clientConfig);
+        client.session.start(function(ks, err){
+            if (err){
+                logger.error("Failed to create session", err);
+                deferred.reject(err);
+            }
+            if(ks){
+                logger.info('session created successfully. ks: ' + ks);
+                client.setSessionId(ks);
+                deferred.resolve();
+            }
+        }, adminSecret, null, kalturaTypes.KalturaSessionType.ADMIN, partnerId, null, null);
+
+        return deferred.promise;
+    };
+
+    var getLiveEntries = function () {
+
+        var deferred = Q.defer();
+        var filter = new kalturaVO.KalturaLiveStreamEntryFilter();
```

Comment:
You can add updatedAtGreaterThanOrEqual to the filter to get only the entires that changed since your last check, it will save processing time and HTTP content length.